src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collections/TimeExpiringSet.java

```diff
@@ -0,0 +1,76 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.collections;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import java.util.Iterator;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import javax.annotation.Nonnull;
+
+/**
+ * Stores a set of elements which are automatically removed from the Set after a given time period.
```

Comment:
nit: "Stores a set of elements which are automatically removed from the Set after a given time period.", can we update this to "Caches a set of elements which are automatically evicted based on the cache TTL"

Just to keep it consistent with Cache terminologies.

Reply:
Done